docs/dev/running-test.md

```diff
@@ -1,64 +1,97 @@
 # How to run Assisted-service subsystem tests
 
-Assisted-service subsystem tests require deploying the assisted-service on a k8s cluster together with DB and storage services.
-The subsystem tests are located on the [subsystem](https://github.com/openshift/assisted-service/tree/master/subsystem) directory.
+There are two "flavors" of subsystem tests:
 
-## Subsystem tests pre-configuration
+* subsystem tests for the service deployed in REST-API mode
+* subsystem tests for the service deployed in kube-api mode
 
-* Run [minikube](https://minikube.sigs.k8s.io/docs/start/) on your system
-* Enable [registry addon](https://minikube.sigs.k8s.io/docs/handbook/registry/) on your minikube:
+Assisted-service subsystem tests require that you first deploy the
+assisted-service on a k8s cluster (e.g. minikube).
+
+This document will explain how you can easily deploy the service in preperation
+for subsystem tests, and also how to run the tests themselves.
+
+The subsystem tests themselves are located in the
+[subsystem](https://github.com/openshift/assisted-service/tree/master/subsystem)
+directory, but they are launched via `make` targets in the Makefile at the root
+of this repository.
+
+## Service deployment
+
+This section will show you how you can deploy the assisted installer to
+a minikube cluster in preperation for subsystem tests.
+
+### minikube
+
+First we must prepare the minikube cluster -
 
 ```bash
+# Optionally delete the existing minikube cluster:
+# minikube delete
+
 minikube start
+
+# Start the minikube container registry and make it accessible locally:
 minikube addons enable registry
+nohup kubectl port-forward svc/registry 5000:80 -n kube-system &>/dev/null &
+export LOCAL_SUBSYSTEM_REGISTRY=localhost:5000
+
+echo "Waiting for registry to become ready..."
+while ! curl --location $LOCAL_SUBSYSTEM_REGISTRY; do
+    sleep 10
+	echo "kubectl registry service tunnel at port 5000 is not available yet, retrying..."
```

Comment:
seems like this one has specifically tab or something, so indentation seems weird